### PR TITLE
Remove Babel 6 from internal dependencies

### PR DESCRIPTION
**Pull request by rekmarks**
This PR updates our internal dependencies to remove `babel@6` from our internal dependency graph. The only dependency requiring `babel@<7` or `babelify@<10` now is `ganache-core`, a dev dependency.

Everyone developing on/with the extension should see less build issues, and we should be able to use `yarn/npm link` now, or for a time.

The affected packages (and the corresponding PRs) are:
1. https://github.com/MetaMask/obs-store/pull/8
2. https://github.com/MetaMask/json-rpc-engine/pull/18
3. https://github.com/MetaMask/eth-json-rpc-middleware/pull/12
4. https://github.com/MetaMask/eth-json-rpc-infura/pull/13
5. https://github.com/MetaMask/eth-json-rpc-filters/pull/9
6. https://github.com/MetaMask/eth-block-tracker/pull/41
7. https://github.com/MetaMask/web3-provider-engine/pull/318
8. https://github.com/MetaMask/KeyringController/pull/32
9. https://github.com/MetaMask/eth-token-tracker/pull/13
10. https://github.com/MetaMask/metamask-inpage-provider/pull/5
11. https://github.com/MetaMask/gaba/pull/145
12. https://github.com/kumavis/web3-stream-provider/pull/3

All PRs are merged, all packages

### CODE REVIEW

**Comment by rekmarks** [comment]
Currently failing e2e tests. 

**Chrome:** throws on this line of `web3-stream-provider`: https://github.com/kumavis/web3-stream-provider/blob/master/index.js#L54

**Firefox:** `Timeout Error` with reference to `TimeoutError: Waiting for element to be located By(css selector, .unlock-page__link--import)`

**Comment by kumavis** [comment]
@rekmarks this could be caused by non-unique request ids if they are not going through an id-remapper

multiple requests with matching ids start, response for one [deletes the stored callback](https://github.com/kumavis/web3-stream-provider/blob/6e87ed0943efca7e6ebb8f5afd65ca6429fecad3/index.js#L55), then the second response comes in and the callback is missing

potential solutions:
- incomplete: validate that if a request comes in for an id when we're already waiting on that id, through at request time.
- better: ensure we're using an id-remapper
- alternate: drop web3 stream provider for something like dnode that tracks request response for us

**Comment by Gudahtt** [comment]
I'm seeing this error locally as well, on both Chrome and Firefox.

Maybe one of these dependency changes is causing this duplicate request id issue? It wasn't clear to me what might have triggered this change.

**Comment by rekmarks** [comment]
@Gudahtt I'm now able to repro as well. The `Timeout Error` appears to result from the `web3-stream-provider` error. Currently testing the hypothesis that we need to put an `idRemapMiddleware` somewhere.

**Comment by rekmarks** [comment]
I believe I've identified the RPC requests with the duplicate IDs: https://github.com/MetaMask/eth-block-tracker/blob/master/src/polling.js#L69

Clearly these requests are passed through the extension UI's `StreamProvider` here: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/ui.js#L129-L142

The question is, how are we processing these requests? I can't find the RpcEngine for the background connection on the UI-side. Adding an `idRemapMiddleware` to the background's provider connection middleware stack (see [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.js#L1406-L1438) and [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.js#L1378-L1404) and [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.js#L1313-L1329)) does nothing as the error is occurring in the UI.

**Comment by rekmarks** [comment]
Added id remapping to `web3-provider-stream`, solving the duplicate id error: https://github.com/rekmarks/web3-stream-provider

Now something else broke, and upon review, my suspicion is that transactions are being created with undefined IDs, given that this error is thrown: https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/ducks/confirm-transaction/confirm-transaction.duck.js#L353

Specifically, I think our now JSON RPC 2.0-compliant `json-rpc-engine` is escalating RPC errors, breaking existing workflows.

**Comment by rekmarks** [comment]
Ready for review.

e2e tests are passing with the same errors as for https://github.com/MetaMask/metamask-extension/pull/7029

Still needs QA @tmashuang 

@whymarrh, I don't want to mess with your RC, but?

File: package.json [hunk 1]
```diff
@@ -161,7 +161,7 @@
     "textarea-caret": "^3.0.1",
     "valid-url": "^1.0.9",
     "web3": "^0.20.7",
-    "web3-stream-provider": "^3.0.1",
+    "web3-stream-provider": "rekmarks/web3-stream-provider#id-remap",
```
Comment (whymarrh) [inline package.json]: We should upstream this so we're not using a Git URL with a branch reference

File: ui/app/pages/confirm-transaction/confirm-transaction.component.js [hunk 2]
```diff
@@ -70,7 +65,8 @@ export default class ConfirmTransaction extends Component {
     if (isTokenMethodAction) {
       getTokenParams(to)
     }
-    this.props.setTransactionToConfirm(transactionId || paramsTransactionId)
+    const txId = transactionId || paramsTransactionId
+    if (txId) this.props.setTransactionToConfirm(txId)
```
Comment (whymarrh) [inline ui/app/pages/confirm-transaction/confirm-transaction.component.js]: Why is this needed?

File: yarn.lock [hunk 3]
```diff
@@ -22683,12 +22626,12 @@ web3-shh@1.0.0-beta.35:
     web3-core-subscriptions "1.0.0-beta.35"
     web3-net "1.0.0-beta.35"
 
-web3-stream-provider@^3.0.1:
+web3-stream-provider@rekmarks/web3-stream-provider#id-remap:
   version "3.0.1"
-  resolved "https://registry.yarnpkg.com/web3-stream-provider/-/web3-stream-provider-3.0.1.tgz#f5a593a8eefe808f85eb5fb1f344e5838050f814"
-  integrity sha1-9aWTqO7+gI+F61+x80Tlg4BQ+BQ=
+  resolved "https://codeload.github.com/rekmarks/web3-stream-provider/tar.gz/1fd86766630d407686efba51b1d14c49de62b212"
```
Comment (whymarrh) [inline yarn.lock]: This Git URL isn't ideal

File: package.json (hunk 1 above)
Comment (whymarrh) [reply package.json]: (As shown below.)

**Review by whymarrh — DISMISSED** [review]
A few comments inline. Also can we rebase this on the latest `develop`?

File: ui/app/pages/confirm-transaction/confirm-transaction.component.js (hunk 2 above)
Comment (rekmarks) [reply ui/app/pages/confirm-transaction/confirm-transaction.component.js]: It gets rid of the following error otherwise thrown on occasion by that call:
`[2019-08-21 20:32:23.690][e2e] chrome-extension://aohcjlklkjjpdgebibknknkinbdpdmoe/ui.js 0:265094 "Transaction with id undefined not found"`

File: package.json (hunk 1 above)
Comment (rekmarks) [reply package.json]: Pending: https://github.com/kumavis/web3-stream-provider/pulls

File: ui/app/pages/confirm-transaction/confirm-transaction.component.js (hunk 2 above)
Comment (rekmarks) [reply ui/app/pages/confirm-transaction/confirm-transaction.component.js]: I thought that was a problem with my changes but that error is currently thrown by the `7.1.0` RC and I think by current prod: https://circleci.com/gh/MetaMask/metamask-extension/102563?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Comment (whymarrh) [reply ui/app/pages/confirm-transaction/confirm-transaction.component.js]: Can we split this out into its own PR?

Comment (rekmarks) [reply ui/app/pages/confirm-transaction/confirm-transaction.component.js]: Done: https://github.com/MetaMask/metamask-extension/pull/7053

File: package.json (hunk 1 above)
Comment (kumavis) [reply package.json]: published as `web3-stream-provider@4.0.0`

Comment (rekmarks) [reply package.json]: Package updated.

File: .gitignore [hunk 4]
```diff
@@ -32,7 +32,7 @@ builds/
 builds.zip
 docs/jsdocs
 
-development/bundle.js
+development/bundle.js*
```
Comment (Gudahtt) [inline .gitignore]: What is this change for? I don't see any files that this change would apply to

Comment (rekmarks) [reply .gitignore]: On review, it was a temp file that was probably left over when I interrupted some script. I amended the commit.

**Review by Gudahtt — APPROVED** [review]
Looks good!
